### PR TITLE
BUGFIX: Crash when accessing nonexist files with file protocol

### LINKED ISSUE #1524: Access to file:// is leaky and can crash the game
- [X] Steps to reproduce
`wget wget file:///c:/bogus_file test.txt`
- [X] Version of the game, e.g. Bitburner v2.3.2dev (64933419d)

What I expect to happen: The same thing that happens on the web (the error reporting could be improved, but it is at least functional):
image
```
[home /]> wget file:///c:/bogus_file test.txt
wget failed: {"stack":"TypeError: Failed to fetch\n    at i (https://bitburner-official.github.io/dist/main.bundle.js:9:752192)\n    at ve.executeCommand (https://bitburner-official.github.io/dist/main.bundle.js:9:716327)\n    at ve.executeCommands (https://bitburner-official.github.io/dist/main.bundle.js:9:712672)\n    at onKeyDown (https://bitburner-official.github.io/dist/main.bundle.js:9:760235)\n    at Object.ze (https://bitburner-official.github.io/dist/vendor.bundle.js:155:7398232)\n    at qe (https://bitburner-official.github.io/dist/vendor.bundle.js:155:7398386)\n    at https://bitburner-official.github.io/dist/vendor.bundle.js:155:7416564\n    at Ei (https://bitburner-official.github.io/dist/vendor.bundle.js:155:7416658)\n    at xi (https://bitburner-official.github.io/dist/vendor.bundle.js:155:7417072)\n    at https://bitburner-official.github.io/dist/vendor.bundle.js:155:7422702","message":"Failed to fetch"}
```

What happens on electron:
image
image
image
Followed by game crash. This is happening in the privileged code I'm pretty sure.

### PR DESCRIPTION
**Pull request by catloversg**
The crash is caused by the usage of `realpathSync`. That code was added in #1427. Technically, we can use `fs.existsSync` to check `filePath` before using it instead of putting our code in a big try-catch block, but I want to make sure that block will never cause a problem again.

When the player tries to access files with the `file://` protocol in the Electron app and those files violate our constraints ("Only allow access to files in "dist" folder or html files in the same directory"), we return the content of `fileError.html`. This is the expected behavior.

Fixes #1524.

### CODE REVIEW

**Comment by d0sboots** [comment]
I think probably we should do the same behavior as the browser, and return an access violation error to the user-side. But that's a bigger change, and this restores the previous behavior.

File: electron/main.js [hunk 1]
```diff
@@ -201,13 +201,20 @@ global.app_handlers = {
...
-    const realPath = realpathSync(filePath);
-    const relativePath = path.relative(__dirname, realPath);
-    // Only allow access to files in "dist" folder or html files in the same directory
-    if (method === "GET" && (relativePath.startsWith("dist") || relativePath.match(/^[a-zA-Z-_]*\.html/))) {
-      callback(realPath);
-      return;
+    let filePath;
+    let realPath;
+    let relativePath;
+    try {
```
Comment (d0sboots) [inline electron/main.js:212]: Let's add a comment here about the known cases where this code can throw (and why it's extra bad for exceptions to bubble here)

**Comment by catloversg** [comment]
As we discussed on Discord, we will return 403 when the player tries to access invalid files.

Capture

**Comment by d0sboots** [comment]
Looks like the same behavior as web.

**Pull request merged**